PurchasesTests/Purchasing/PurchasesTests.swift

```diff
@@ -1374,6 +1374,20 @@ class PurchasesTests: XCTestCase {
         expect(self.receiptFetcher.receiptDataCalled).to(beTrue())
     }
 
+    func testGetEligibilityDoesntRefreshReceipt() {
+        setupPurchases()
+        self.receiptFetcher.shouldReturnReceipt = false
+
+        expect(self.requestFetcher.refreshReceiptCalled) == false
+        expect(self.receiptFetcher.receiptDataCalled) == false
+
+        purchases!.checkTrialOrIntroductoryPriceEligibility([]) { (eligibilities) in
```

Comment:
The `!` isn't required right? It's already an implicitly unwrapped optional.

Reply:
great catch!! I was mislead because we're actually treating `purchases` as a regular optional everywhere in this file. I updated all references, since that's not necessary